### PR TITLE
feat: adds --test flag in yr fmt

### PR DESCRIPTION
**Pull request by shellcromancer**
Adds -t/--test flag in yr fmt which returns an error code if formatting made changes on the file.

This feature exists in a few other language formatting tools ([jsonnet](https://github.com/google/go-jsonnet/blob/master/cmd/jsonnetfmt/cmd.go#L47-L48), [terraform](https://developer.hashicorp.com/terraform/cli/commands/fmt#check), [gofmt](https://github.com/golang/go/blob/master/src/cmd/gofmt/gofmt.go#L33)) which is useful as part of stylistic CI/CD checks

### CODE REVIEW

**Comment by shellcromancer** [comment]
Signed the Google CLA, should be ready for a maintainer review, thanks! 👍

File: yara-x-cli/src/commands/fmt.rs [hunk 1]
```diff
@@ -18,27 +19,38 @@ pub fn fmt() -> Command {
...
 
-                File::create(file.as_path())?
-                    .write_all(formatted.into_inner().as_slice())?;
+            formatter.format(input.as_slice(), &mut formatted)?;
+            formatted.rewind()?;
+
+            let output = formatted.into_inner();
+
+            if *test.unwrap() && input != output {
+                process::exit(3)
```
Comment (plusvic) [inline yara-x-cli/src/commands/fmt.rs:47]: Why not `process::exit(2)`? Exit code 2 is not used for any purpose yet. 

I'm also wondering if it wouldn't be better to return the path of the files that requires changes, as `gofmt -l` does. This provides more information to the user. If you call `yr fm` with multiple files, you may want to know which files are changed and which don't.

Comment (shellcromancer) [reply yara-x-cli/src/commands/fmt.rs]: Thank you for taking a look! I updated this to use exit code 2, and output a list of files that require changes. 

image

**Comment by plusvic** [comment]
In https://github.com/VirusTotal/yara-x/commit/4b0889938df1feff849e6c9cdd4d84ab59eeec86 I made multiple changes to the `fmt` command. Now it always exits with error code 0 if files didn't change or 1 if otherwise.

**Comment by shellcromancer** [comment]
Makes sense, thanks for including the feature in your changes. 👍

**Pull request closed without merging**